homeassistant/util/dt.py

```diff
@@ -316,7 +316,7 @@ def _lower_bound(arr: list[int], cmp: int) -> int | None:
         # clocks are rolled back. In this case, we want to trigger
         # on both the DST and non-DST time. So when "now" is in the DST
         # use the DST-on time, and if not, use the DST-off time.
-        fold = 1 if now.dst() else 0
+        fold = 0 if now.dst() else 1
         if result.fold != fold:
             result = result.replace(fold=fold)
```

Comment:
Are there timezones with multiple DST changes? If so, that would break.

I think better would be just

```suggestion
        result = result.replace(fold=now.fold)
```

Reply:
Do you mean if fold can hold other values than 0 or 1?

Reply:
Fold should only be 0 or 1 https://www.python.org/dev/peps/pep-0495/#why-not-call-the-new-flag-isdst

Reply:
No i mean fold is always 0 or 1 sure. But what if you have multiple DST levels in one year. like for some months +1h, others +2h and in winter +0h. I know morrocco had some really weird DST stuff for a while

Reply:
Update: here's a link https://en.m.wikipedia.org/wiki/Daylight_saving_time_in_Morocco

In 2012&13 they had two DST phases (during summer EXCEPT during Ramadan) 🧐

Looks like there were not two "levels" of DST though. My concern was if there's a fold from dst=2h to dst=1h. There settings the fold directly would be more safe